### PR TITLE
Add a blog for ocitools generate library

### PR DESCRIPTION
**Pull request by hmeng-19**
Signed-off-by: Haiyan Meng <hmeng@redhat.com>

### CODE REVIEW

**Comment by hmeng-19** [comment]
@mrunalp , @rhatdan , PTAL.

File: source/blog/2016-08-01-ocitools-libgen.html.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+using the [ocitools generate library](https://github.com/opencontainers/ocitools/tree/master/generate).
+
+There are four steps to create an OCI configurations using the ocitools generate library.
+
+1. Import the ocitools generate libraray into your project;
+2. Create an OCI specification generator;
+3. Modify the specification by calling different methods of the specification generator;
+4. Save the specification.
+
+READMORE
```
Comment (mrunalp) [inline source/blog/2016-08-01-ocitools-libgen.html.md]: Is this meant to be a link?

**Comment by mrunalp** [comment]
Looks fine overall  :+1:

File: source/blog/2016-08-01-ocitools-libgen.html.md (hunk 1 above)
Comment (hmeng-19) [inline source/blog/2016-08-01-ocitools-libgen.html.md]: This is just a mark to limit the amount of the content to show by default. It automatically functions as an expansion button.

**Comment by jberkus** [comment]
sorry for delay, currently at Flock.  Should be able to post this sometime tommorrow, assuming no issues.

**Comment by jberkus** [comment]
@hmeng-19

looks fine, except that the blog post doesn't explain what an OCI generator is *for*.  What is it?  What do I use it for?

Please update the PR or explain in the comments, thanks.

**Comment by hmeng-19** [comment]
@jberkus , thanks for pointing out. 
I updated the `Overview` section of the blog to explain the Generator:
```
Different methods are defined on the Generator type to allow the user to customize the specification pointed by the Generator.
```
I also updated the date of the blog from `Aug 1st` to `Aug 5th`. PTAL.

**Comment by jberkus** [comment]
Right, but what is an OCITools generator *for*?  Why would I want one?  What real-world problem does having an OCI bundle actually solve?  Who is this for?

I mean, I know a bit about runC, and a lot about Docker, and I can't figure out why I would want to use this code.  The blog post should have at least 1-2 lines which explains this.

**Comment by jberkus** [comment]
Note that this would be less necessary if the linked repo had documentation which explained its purpose in general terms, but it doesn't.

**Comment by jberkus** [comment]
Also, if you're having trouble with wording for the explaination, try explaining here and I'll polish it.

**Comment by rhatdan** [comment]
OCI runc is a cool new tool for running containers on Linux machines.  It follows the OCI container runtime specification.  As of docker-1.11 it is the main mechanism that docker uses for launching containers.

The really cool thing is that you can use runc without even using docker.  Basically you get a rootfs on your disk. (A rootfs is a directory that includes all of your software and usually follows the basic layout of `/`. Their are several tools that can create a rootfs, including dnf or the atomic command.  Once  you have a rootfs, you need to create a config.json file which runc will read.  config.json has all of the specifications for running a container, things like which namespaces to use, which capabilities to use in your container, what is the pid 1 of your container etc.  It is somewhat similar to the output of docker inspect.  

### ocitools

Creating and editing the config.json is not for the faint of heart, so we developed a command line tool called `ocitools generate`  that can do the hard work of creating the config.json file. 

...

**Comment by rhatdan** [comment]
Hows that look for an intro.

**Comment by jberkus** [comment]
@rhatdan thanks!  I'll add it in and merge.

**Comment by hmeng-19** [comment]
@jberkus , I can copy the content Dan wrote into the PR. Did you already do it?

**Comment by hmeng-19** [comment]
@rhatdan , do you mind I add your name into the author list of the blog?

**Comment by hmeng-19** [comment]
@jberkus , PTAL.
I added the content Dan wrote into the PR and updated the time.

**Comment by hmeng-19** [comment]
@rhatdan , PTAL.

**Comment by rhatdan** [comment]
Fine with me.

**Comment by jberkus** [comment]
Created as a new branch, introduction and corrections added, and pushed to master.  Will come up on the site once the automation picks it up.

**Comment by jberkus** [comment]
@hmeng-19 sorry, I was working on this in parallel. Already pushed it.

**Comment by hmeng-19** [comment]
@jberkus , it is okay. Glad it worked well eventually and got merged. Thanks.

**Pull request closed without merging**